hudi-client/hudi-client-common/src/main/java/org/apache/hudi/metadata/HoodieBackedTableMetadataWriter.java

```diff
@@ -843,28 +840,37 @@ protected void cleanIfNecessary(BaseHoodieWriteClient writeClient, String instan
   }
 
   /**
-   * This is invoked to bootstrap metadata table for a dataset. Bootstrap Commit has special handling mechanism due to its scale compared to
+   * This is invoked to initialize metadata table for a dataset. Bootstrap Commit has special handling mechanism due to its scale compared to
    * other regular commits.
-   *
    */
-  protected void bootstrapCommit(List<DirectoryInfo> partitionInfoList, String createInstantTime) {
-    List<String> partitions = partitionInfoList.stream().map(p ->
-        p.getRelativePath().isEmpty() ? NON_PARTITIONED_NAME : p.getRelativePath()).collect(Collectors.toList());
-    final int totalFiles = partitionInfoList.stream().mapToInt(p -> p.getTotalFiles()).sum();
+  private void initialCommit(String createInstantTime) {
```

Comment:
+1 for extracting the actual commit part out of bootstrap (initialize filegroup). This will help me for async indexing.
How about adding partitionType as another parameter and making `initialCommit` part of the `HoodieTableMetadataWriter` interface? That way we can call from write client or any action executor if needed. Not necessary to do this in PR but just a suggestion to consider.